### ISSUE TITLE
Local variable are not displayed in the Code snippet after running the request

### ISSUE THREAD

**Issue by premkumartg89** [a85/POSTMan-Chrome-Extension#10258]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the Issue

Code snippet is not showing the local variable which was used in body.

### Steps To Reproduce

1. Open new request.
2. Click on Pre-request Script
3. Declare pm.variable.set("code", "100")
4. Use the variable "code" in the body.
5. Run the request.
6. Open Code snippet.
7. Verify the "Code" value.

Expected Result: Code value should be displayed.

Actual Result: Variable name is displayed instead of value.

### Environment Information

```markdown
- Operating System: Win10
- Platform Type: Native App
- Postman Version: 8.11.1
```

**Comment by xk0der** [comment]
@premkumartg89 I was unable to replicate the issue on MacOS, with Native app version 8.12.1

<img width="1000" alt="Screen Shot 2021-09-07 at 11 42 12 PM" src="https://user-images.githubusercontent.com/21366/132338890-3255df9d-094c-45ac-80bf-08944bcc9658.png">

<img width="579" alt="Screen Shot 2021-09-07 at 11 42 50 PM" src="https://user-images.githubusercontent.com/21366/132338962-85322d1c-02b4-4406-93ab-5c34aac3863e.png">


* Which language did you select when generating the code?
* What payload type you selected for your request? (raw/form-data etc)


Do add screenshots showing the full request URL and method along with body-payload.
Also include the code snippet pane's screenshot to show the issue.

A screen recording showing the replication steps would be even better.

**Comment by xk0der** [comment]
Closing as there was no further follow-up!

**Comment by Mike-Soft** [comment]
@xk0der It was clearly stated in the issue description that the issue was found on Windows. How can you close issue after checking on Mac? Unfortunately after 2 years there is still the same problem. If I create local variable and use in in request body the code snippet showing what was sent is showing variable name instead of a value. When I use console.log(pm.request.body); in "pre-req." script I also see variable name, when using console.log from "Tests" tab I can finally see the value.

**Comment by xk0der** [comment]
@Mike-Soft This was closed out as we were pending on a follow-up from the requestor to gather more information on what was happening on a Windows machine. If you are facing a similar issue, please [do log a new one](https://github.com/postmanlabs/postman-app-support/issues/new/choose) so that we can have a look again 👍 (Make sure to include the items I had requested in my [comment above](https://github.com/postmanlabs/postman-app-support/issues/10258#issuecomment-914233159))

**Comment by premkumartg89** [comment]
@amit i didn't respond as you response doesn't made sense to me, i clearly
wrote it happens in windows but you tried in Mac and again asking
additional steps from windows, and then I thought even though I provide
additional details you will test on Mac so left as it is 😀.

**Comment by xk0der** [comment]
@premkumartg89 appreciate the follow-up :) 

@premkumartg89 @Mike-Soft 

I had another look at this issue and I stand corrected.

Since local variables are _only_ available during the instance when the request is being executed - these won't be available when you are generating the code. (As the variable and thus its value are no longer available)

The code generator can only resolve variables which persist - so only Collection and Environment variables would work - with whatever values these variables currently store when the code is generated.

Just to be absolutely sure, I tested on both Mac and a Windows machine this time - and things work as expected when using either Collection or Environment variable.

_My pre-request looks as follows_
```
pm.collectionVariables.set("code", "100");
```

_Body is set to raw-JSON_
```
{
     "something": "{{code}}"
}
```

Generated Code - AFTER sending the request to make sure collection-variable was set.
```
curl --location --request GET 'postman-echo.com/get' \
--header 'Content-Type: application/json' \
--header 'Cookie: sails.sid=XYZ' \
--data '{
    "something": "100"
}'
```

I hope this clarifies the behaviour.

----
Why it worked for me the first time I tested? The only possible thing that comes to mind is that at one point during testing I may have tested using a collection or an environment variable and when I generated the code - I guess that was being used. 😬

**Comment by Mike-Soft** [comment]
@xk0der Thanks for the response. This still looks like a problem to me. 
My understanding is that the code snippet generated on the right should already contain the code generated with the values? Otherwise how can someone debug or export the code that was used when the values are not shown there? 
I will create a new issue on that topic.
